feature-extensions/splunk/setup.py

```diff
@@ -0,0 +1,142 @@
+    )
+    mainParser.add_argument(
+        '--scm-password',
+        help='Specify the scm password'
+    )
+    mainParser.add_argument(
+        '--scm-pathext',
+        help='Specify the scm repo path ext (Use "scm" for bitbucket)'
+    )
+
+    args = mainParser.parse_args()
+    args.func(args)
+
+
+def install(args):
+    print(
+        colors.OKGREEN +
+        "\nThis will install {0} functionality into your Jazz deployment.\n".format(featureName)
+        + colors.ENDC)
+
+    configureSplunk(args, True)
+
+
+def configureSplunk(args, splunk_enable):
+
+    if not args.splunk_endpoint:
+        args.splunk_endpoint = raw_input("Please enter the Splunk Endpoint: ")
+
+    if not args.splunk_token:
+        args.splunk_token = raw_input("Please enter the Splunk Token: ")
+
+    if not args.splunk_index:
+        args.splunk_index = raw_input("Please enter the Splunk Index: ")
+
+    if not args.scm_repo:
+        args.scm_repo = raw_input("Please enter the SCM Repo: ")
+
+    if not args.scm_username:
+        args.scm_username = raw_input("Please enter the SCM Username: ")
+
+    if not args.scm_password:
+        args.scm_password = raw_input("Please enter the SCM Password: ")
+
+    if not args.scm_pathext:
+        args.scm_pathext = raw_input("Please enter the Splunk Pathext (Use \"/scm\" for bitbucket): ") or "/"
+
+    if not args.splunk_endpoint and not args.splunk_token and not args.splunk_index:
+        print(colors.FAIL +
+              'Cannot proceed! No install possible'
+              + colors.ENDC)
+        return True
+
+    replace_config(args)
+
+
+def replace_config(args):
+
+    # Clone the SCM
+    subprocess.check_call(
+        [
+            "git",
+            "clone",
+            ("http://%s:%s@%s%s/slf/jazz-build-module.git") %
+            (args.scm_username,
+             urllib.quote(
+                 args.scm_password),
+             args.scm_repo,
+             args.scm_pathext),
+            "--depth",
+            "1"])
+
+    configFile = "jazz-installer-vars.json"
+    buildFolder = './jazz-build-module/'
+    # Read in the file
+    with open(buildFolder+configFile, 'r') as file:
+        filedata = file.read()
+
+    # Replace the target string
+    filedata = filedata.replace('"{ENABLE_SPLUNK_FEATURE}"', 'true')
+    filedata = filedata.replace('{SPLUNK_ENDPOINT}', args.splunk_endpoint)
+    filedata = filedata.replace('{SPLUNK_TOKEN}', args.splunk_token)
+    filedata = filedata.replace('{SPLUNK_INDEX}', args.splunk_index)
+
+    # Write the file out again
+    with open(buildFolder+configFile, 'w') as file:
+        file.write(filedata)
+
+    # Commit the changes
+    subprocess.check_call(["git", "add", configFile], cwd=buildFolder)
+    subprocess.check_call(["git", "commit", "-m", "'Adding Splunk feature'"], cwd=buildFolder)
+    subprocess.check_call(["git", "push", "-u", "origin", "master"], cwd=buildFolder)
+    subprocess.check_call(["sudo", "rm", "-rf", buildFolder])
```

Comment:
Remove this `sudo`, it shouldn't be necessary since the files are being cloned by the current user to begin with.

Also for something this simple use built-in file routines to recursively delete the folder, rather than dropping back into the shell.

Reply:
The changes came as part of this PR: https://github.com/tmobile/jazz-installer/pull/361. Since this is a release PR, I have raised a PR for the feature branch here: https://github.com/tmobile/jazz-installer/pull/411. Please take a look.

Reply:
Gotcha, approved and merged.